### ISSUE TITLE
Varying Offence Stat

### ISSUE THREAD

**Issue by Revenate39** [OpenThrone/OpenThrone#272]
I've only noticed today, but my offence stat is all over the place, whilst having not changed unit numbers/types, or purchased/sold items.

Attack was on 09/11/24
![image](https://github.com/user-attachments/assets/9bd1a5fb-df69-4481-a437-2e78b9eedbf7) 
Attack was on 14/11/24
![image](https://github.com/user-attachments/assets/5915faee-9bd1-4b06-be66-5e58a2ddf48c)
Attack was on 14/11/24
![image](https://github.com/user-attachments/assets/3699a58a-77cd-4ffa-ab67-221dfe2d81a1)

**Comment by Revenate39** [comment]
And now appears to be back to 21,351,000

**Comment by uaktags** [comment]
Hmm, interesting. The amount of damage is supposed to come directly from the db, to remove any on demand calculations that would normally cause this issue. To be clear, these are all different attacks, but with the same units, items, etc employed? If so, can you provide the battle ids?

**Comment by camarao69** [comment]
Hello this happened to me to 15 days ago in the batlle ID 10537,  my attack was downgraded to 5 096 216
If you see batle ID 10531 i had at the time an attack of  9 692 336.

I think it was the same problem.

**Comment by Revenate39** [comment]
> Hmm, interesting. The amount of damage is supposed to come directly from the db, to remove any on demand calculations that would normally cause this issue. To be clear, these are all different attacks, but with the same units, items, etc employed? If so, can you provide the battle ids?

Yes all different attacks, all with the same items, troops, upgrades, etc.

Battle ID: 11929
Battle ID: 11883
Battle ID: 11889

**Comment by uaktags** [comment]
Yea it's strange, so the value you're seeing is coming from the db entry, which was captured at the start of the attack. At the end of every attack, we then update both the attacker/defender's values and call it a day. To have different values at the beginning of each attack tells me that the calculation is returning different amounts per run. I'm going to keep playing with this to see where the slow down is, as it feels like a race condition is occurring in the calculation of the units, items, and battleupgrades. There's plenty of conditions and loops, so i'd imagine that could be an area of improvement

**Comment by uaktags** [comment]
I believe i have a fix for this. Ran through a series of tests and it appears that due to references, we actually mess with the user model in memory in some operations. Will be testing in the next few days when i push out the latest patches
